dftimewolf/lib/state.py

```diff
@@ -238,6 +239,10 @@ def StoreContainer(self, container: "interface.AttributeContainer") -> None:
       container (AttributeContainer): data to store.
     """
     with self._state_lock:
+      calling_class = inspect.stack()[1][0].f_locals["self"].__class__.__name__
```

Comment:
Doesn't `self.__class__.__name__` work just fine here? (even if the function is not overridden)

Reply:
Negative. Here we need to look up the stack to find the caller, and identify that class. `self.__class__.__name__` will always give `DFTimewolfState` because that's what `self` is here. 

That said, I've realised we can instead use `inspect.stack()[1][0].f_locals["self"].name` which is the `name` attribute of the module - Giving us the runtime name of the module, when that differs from the class name according to the recipe.

Reply:
... which broke the unit tests of course, because they don't have a `name` attribute. Refactored that to look for a module name and fall back to the class name if it doesn't exist.